controllers/common/common.go

```diff
@@ -320,3 +367,133 @@ func (in *EventLogger) WarningEvent(object runtime.Object, reason string, messag
+	var delta strings.Builder
+	lastParent := "-"
+	for i, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		// Check if the line starts with a "+" or "-" indicating an added or removed configuration.
+		if strings.HasPrefix(trimmedLine, "+") || strings.HasPrefix(trimmedLine, "-") {
+			// Search for the parent and sub-parameters of the line in the given properties.
+			parent := searchParameters(lines, i, parameters)
+			if parent == "param_found" {
+				// If the line represents a parameter itself, add it directly to the delta.
+				trimmedLine := strings.TrimSpace(trimmedLine)
+				line = strings.Join(strings.Fields(trimmedLine), "  ")
+				delta.WriteString(line)
+				delta.WriteString("\n")
+				continue
+			}
+			// If the parent has changed, add a newline and the parent to the delta.
+			if parent != lastParent {
+				delta.WriteString("\n")
+				delta.WriteString(parent)
+				lastParent = parent
+			}
+			// Add the line to the delta.
+			delta.WriteString(trimmedLine)
+			delta.WriteString("\n")
+		}
+	}
+
+	return delta
+}
+
+// searchParameters searches for the parent and sub-parameters of a given line in the given resource properties.
+// It iterates over the lines starting from the given line number and goes upwards to find the relevant information.
+// Parameters:
+//   - lines: A slice of strings representing the lines of the diff.
+//   - lineNumber: The index of the line being processed.
+//   - parameters: A map of resource properties with their corresponding values.
+// Returns:
+//   - A string representing the hierarchy of the parent and sub-parameters, or "param_found" if the line represents a parameter itself.
+//     The hierarchy is constructed in the format: "parent:\n\t"sub-parameter":\n".
+func searchParameters(lines []string, lineNumber int, parameters map[string]interface{}) string {
+	var result string
+
+	for i := lineNumber; i >= 0; i-- {
+		line := lines[i]
+
+		// Check if the line matches any parameter or sub-parameter in the system properties.
+		for param, subParams := range parameters {
+			if subParams != nil {
+				subParamsList, ok := subParams.([]string)
+				if ok {
+					for _, subParam := range subParamsList {
+						// If a sub-parameter is found in the line, construct the corresponding hierarchy.
+						if strings.Contains(line, subParam) {
+							if i == lineNumber {
+								// If the line represents the sub-parameter itself, return the parent parameter.
+								return fmt.Sprintf("%s:\n", param)
+							}
+							// Append the sub-parameter to the hierarchy.
+							result += fmt.Sprintf("%s:\n\t\"%s\":\n", param, subParam)
+							return result
+						}
+					}
+				}
+			}
+		}
+
+		// Check if the line matches any parameter in the system properties.
+		for param := range parameters {
+			if strings.Contains(line, param) {
+				if i == lineNumber {
+					// If the line represents the parameter itself, indicate it with "param_found".
+					return "param_found"
```

Comment:
Can we create constant for "param_found"?

Reply:
Done.